extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java

```diff
@@ -183,6 +183,12 @@ public static enum DeploymentResourceKind {
     @ConfigItem
     Optional<List<String>> imagePullSecrets;
 
+    /**
+     * Enable generation of image pull secret, when credentials are available.
+     */
+    @ConfigItem(defaultValue = "true")
+    boolean imagePullSecretGenerate;
```

Comment:
About the name, I would rename it to `generate-image-pull-secret` instead to follow the same pattern as others properties like `quarkus.kubernetes-client.generate-rbac`.